src/chatty/util/history/HistoryManager.java

```diff
@@ -0,0 +1,206 @@
+package chatty.util.history;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+
+import chatty.Room;
+import chatty.User;
+import chatty.util.UrlRequest;
+import chatty.util.UrlRequest.FullResult;
+
+import chatty.util.irc.MsgTags;
+import chatty.util.settings.Settings;
+
+
+/**
+ * History Manager which should be the entry point for getting historic Chat messages from external services.
+ * Currently, only robotty https://recent-messages.robotty.de is implemented
+ * @author m00hlti
+ */
+public class HistoryManager {
+    private enum MessageType {
+        UNDEFINED,
+        PRIVATMESSAGE,
+        CLEARCHAT
+    }
+
+    private final Settings settings;
+
+    private final static String strHistoryURL = "https://recent-messages.robotty.de/api/v2/recent-messages/";
+    private final int nDefaultLimitMessages = 30;
+    private long lTimeStampBefore = 0;
+    private long lTimeStampAfter = 0;
+
+    private final static String  strRegexRobotty= "display-name=([^;|^\\s]*)[;|\\s]|" +  /* Display Name */
+                                                  ":[\\S]+\\s+PRIVMSG\\s+#[\\S]+\\s+:?([^\\n]*)|" + /* Message */
+                                                  "rm-received-ts=([^;|^\\s]*)[;|\\s]|" + /* TimeStamp */
+                                                  "color=([^;|^\\s]*)[;|\\s]|" /* Color */;
+    private Pattern patRegexRobotty;
+
+    /**
+     * Default Constructor
+     */
+    public HistoryManager(Settings settings) {
+        this.settings = settings;
+        this.patRegexRobotty = Pattern.compile(strRegexRobotty, Pattern.CASE_INSENSITIVE | Pattern.UNIX_LINES) ;
+    }
+
+    /**
+     * Checks if a channel is on the exclusion list
+     * @param channel Channel which should be checked
+     * @return false if not excluded, true if
+     */
+    public Boolean isChannelExcluded(String channel) {
```

Comment:
Minor thing, but this should be returning `boolean` instead of a `Boolean` object.